scripts/irods/test/test_iadmin.py

```diff
@@ -1823,6 +1823,38 @@ def test_iadmin_rmuser_removes_associated_acls__issue_7778(self):
             self.admin.run_icommand(["irmdir", test_collection_name])
             self.admin.run_icommand(["irm", "-f", test_dataobj_name])
 
+    def test_modzone_does_not_accept_invalid_zone_names__issue_7722(self):
+        # Show that "iadmin modzone name" no longer accepts invalid zone names.
+        invalid_zone_name = 'überZone'
+        ec, _, _ = self.admin.assert_icommand(
+            ['iadmin', 'modzone', self.admin.zone_name, 'name', invalid_zone_name],
+            'STDERR',
+            ['-130000 SYS_INVALID_INPUT_PARAM'],
+            input='yes')
```

Comment:
Although I admire the level of trust, I wonder whether we should avoid modifying the zone which is running the tests. If a regression is introduced, there's no recovering from this outside of modifying the database. It's not as big of a deal as in the past since all of our testing is done in relatively short-lived containers now, but I'm wondering whether creating a zone with a valid name and modifying that zone would be better. Up to you

Reply:
That's a good point. The problem is that working with a remote zone doesn't trigger the code path we're interested in.

Your comment does point out the lack of a test for modifying the name of remote zones. I'll add a test for that.

I also think the implementation for renaming zones should be reviewed and made more atomic and recoverable. I'll open an issue for that.

Reply:
Added a new test for renaming a remote zone.

Will test it shortly.